src/js/structs/__tests__/JobTree-test.js

```diff
@@ -0,0 +1,115 @@
+let Job = require('../Job');
+let JobTree = require('../JobTree');
+
+describe('JobTree', function () {
+
+  describe('#constructor', function () {
+
+    beforeEach(function () {
+      this.instance = new JobTree({
+        id: '/group',
+        items: [
+          new JobTree({id: '/group/foo', items: []}),
+          {id: '/group/bar', items: []},
+          {id: '/group/alpha'},
+          new Job({id: '/group/beta'}),
+          {id: '/group/wibble/wobble'}
+        ],
+      });
+    });
+
+    it('defaults id to root tree (groups) id', function () {
```

Comment:
Let's try to have human readable descriptions 😄 . How about `it(creates the root group if no items are supplied)`?